### PR TITLE
url unique validation in service layer

### PR DESCRIPTION
**Pull request by ghost**
# Description

(Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.)

## Type of change

(Please delete options that are not relevant.)

- [ ] Link to a **GH Issue** or **Jira** task here for reference.
- [ ] Bug fix
- [x] New feature

# How Has This Been Tested?

(Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration)

- [x] Tested by myself
- [ ] Unit test

# Developer Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# Review check list:

- [ ] Separation of Concerns followed (Reviewer name/NaN)
- [ ] Code is in sync with existing code patterns/technologies. (Reviewer name/NaN)
- [ ] Are there any obvious optimizations that will improve performance? (Reviewer name/NaN)
- [ ] Can any logging or debugging code be removed? (Reviewer name/NaN)
- [ ] Are all data inputs checked (for the correct type, length, format, and range) and encoded? (Reviewer name/NaN)
- [ ] Are invalid parameter values handled? (Reviewer name/NaN)
- [ ] Is error handling done the correct way? (Reviewer name/NaN)
- [ ] Are all the edge cases (null, 0, negative) handled properly? (Reviewer name/NaN)

### CODE REVIEW

File: packages/headless/src/database/mongodb/branch/branch.model.ts [hunk 1]
```diff
@@ -10,7 +10,10 @@ const BranchAddressSchema = new Schema<BranchAddress>(
             index: true,
         },
         addressLine1: String,
-        addressLine2: String,
+        addressLine2: {
+            type: String,
+            default: ''
```
Comment (ismail-bs) [inline packages/headless/src/database/mongodb/branch/branch.model.ts]: please remove the default value.

File: packages/headless/src/database/mongodb/branch/branch.model.ts [hunk 2]
```diff
@@ -43,6 +46,16 @@ const BranchSchema = new Schema<Branch>(
             type: Boolean,
             default: true,
         },
+        image: {
+            type: String,
+            default: ''
```
Comment (ismail-bs) [inline packages/headless/src/database/mongodb/branch/branch.model.ts]: same as before.

File: packages/headless/src/database/mongodb/branch/index.ts [hunk 3]
```diff
@@ -8,7 +8,14 @@ import { BranchModel } from './branch.model';
 @Injectable()
 export class BranchDatabase implements IBranchDatabase {
     async createBranch(branch: CreateBranchRequest): Promise<Branch | null>{
-        return await BranchModel.create(branch);
+      let newBranch;
+      try {
+        newBranch = await BranchModel.create(branch);
```
Comment (ismail-bs) [inline packages/headless/src/database/mongodb/branch/index.ts]: you may return just one line.
like: 
` return await BranchModel.create(branch);`

File: packages/headless/src/database/mongodb/branch/index.ts [hunk 4]
```diff
@@ -20,25 +27,35 @@ export class BranchDatabase implements IBranchDatabase {
         }
     };
 
+    async getURL(url: string): Promise<boolean> {
+      try {
+        const urlExists = await BranchModel.findOne({ url });
```
Comment (ismail0946) [inline packages/headless/src/database/mongodb/branch/index.ts:32]: getURL & getStore functions are the same strategy. So you may check getStore by passing url in service layer.

File: packages/headless/src/database/mongodb/branch/index.ts [hunk 5]
```diff
@@ -20,25 +27,35 @@ export class BranchDatabase implements IBranchDatabase {
...
+        if (urlExists) return true;
```
Comment (ismail-bs) [inline packages/headless/src/database/mongodb/branch/index.ts]: `return urlExists ? true : false;`

File: packages/headless/src/database/mongodb/branch/index.ts [hunk 6]
```diff
@@ -20,25 +27,35 @@ export class BranchDatabase implements IBranchDatabase {
...
         }
-        
+
     }
 
     async getBranch(branchId: string): Promise<Branch | null>{
-        try{        
+        try{
             const branch = await BranchModel.findOne({id: branchId});
             console.log(branch)
```
Comment (ismail-bs) [inline packages/headless/src/database/mongodb/branch/index.ts]: remove the log. & you may return just one line.

`return await BranchModel.findOne({id: branchId});`

File: packages/headless/src/modules/branch/services/index.ts [hunk 7]
```diff
@@ -28,6 +28,14 @@ export class BranchService{
             errors: null,
             code: HttpStatus.BAD_REQUEST,
           }
+          
+        const urlExists = await this.branchRepo.getURL(branch.url);
```
Comment (ismail0946) [inline packages/headless/src/modules/branch/services/index.ts:32]: remove the function..you may use for this .
await this.branchRepo.getStore({url})

**Pull request merged**